airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/catalog.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+import json
+import copy
+from typing import List
+
+from dagster import asset, OpExecutionContext
+
+from ..utils.dagster_helpers import OutputDataFrame, output_dataframe
+from ..models.metadata import PartialMetadataDefinition
+
+
+GROUP_NAME = "catalog"
+
+# HELPERS
+
+
+def apply_overrides_from_catalog(metadata_data: dict, override_catalog_key: str) -> dict:
+    # Make a deep copy of the input dictionary to avoid side effects
+    metadata_data_copy = copy.deepcopy(metadata_data)
```

Comment:
Nit: as you reuse this *deepcopy* pattern you could implement a [decorator](https://realpython.com/primer-on-python-decorators/) to always deepcopy the input dict